### PR TITLE
Convert trait type parameters to associated types

### PR DESCRIPTION
**Pull request by bjz**
rust-lang/rust#24092 has caused quite a bit of ugliness, but I've tried to work around it as best I can.

### CODE REVIEW

**Comment by bjz** [comment]
cc @csherratt @kvark - does this look ok?

**Comment by kvark** [comment]
Could you explain a bit on why you are doing it?

File: src/transform.rs [hunk 1]
```diff
@@ -27,24 +27,24 @@ use vector::*;
 /// A trait representing an [affine
 /// transformation](https://en.wikipedia.org/wiki/Affine_transformation) that
 /// can be applied to points or vectors. An affine transformation is one which
-pub trait Transform<S: BaseNum, V: Vector<S>, P: Point<S, V>>: Sized {
+pub trait Transform<P: Point>: Sized {
```
Comment (kvark) [inline src/transform.rs]: sweet

**Comment by kvark** [comment]
Looked through it, looks nice.

**Comment by bjz** [comment]
This part of the Rust Book is good: https://doc.rust-lang.org/book/associated-types.html

**Comment by bjz** [comment]
Ok, I'm going to merge this.

**Pull request merged**